### DISCUSSION TITLE
Readme split up

### DISCUSSION THREAD

**Discussion by bartbutenaers** [phoddie/node-red-mcu#110] [General]
Hi everybody,

Currently the readme page contains ton of useful information, but I have to admit that it is kind of overwhelming when you want to start with this.  And I am a bit worried that new users will turn there back, because they cannot cope with the large amount of text offered.  Which would be the opposite of what this project deserves...

Instead of putting it all on a single page, I think it would be useful to give every node it's own MD file (in its own directory).  

So my idea was to split it up.  Something like this:

## Install
There are two ways to install this node:
1. Install it by executing following command from within your .node-red folder: 
   ...
   Install it like this when you want to execute the `mcconfig` commands manually.

3. Install the [node-red-mcu-plugin](https://github.com/ralphwetzel/node-red-mcu-plugin) sidebar for the Node-RED editor, which will automatically install this library as a depedency.  Install it like this when you want to program your MCU via the sidebar, and let the sidebar generate all `mcconfig` commands under the cover.

## How this node works
Some short intro about Moddable, and so on ...

## Supported nodes
Normally most Node-RED nodes can be converted automatically to run on your MCU.  However take into account that some of them might be too resource intensitive to run on an MCU and blabla...

Some dedicated nodes:

### Core nodes
Some of the Node-RED core nodes will get automatically customized Javascript backend code, that allows to run them on an MCU:
   + Http-request
   + Inject
   + Change
   + ...

Some extra nodes are available which only run on an MCU, not on a NodeJs backend:
   + OTA update
   + Sensor
   +

**Comment by bartbutenaers** [comment]
Hopefully you like the idea.  Now I am off to the daily job...

**Reply by phoddie** [reply]
@bartbutenaers – I completely agree with you that this repository's readme needs to be revisited, and definitely broken into pieces. It began as a description of my experiments to try to get Node-RED working at all on an MCU. The experiment was a success, so now it is time to rethink the readme. I want to complete the current project of making the various MCU nodes available through npm and installable through the Node-RED Editor before tackling the readme. That's because having the MCU nodes more easily available should simplify the instructions quite a bit.

Your outline looks like a very reasonable starting point. I'd welcome input from others about what they think is essential at the top-level, especially for developers just getting started.

**Reply by bartbutenaers** [reply]
BTW somewhere in the top of your page, I would add a link to a quickstart guide (something similar to our [tutorial](https://discourse.nodered.org/t/tutorial-read-a-sensor-by-running-a-node-red-flow-on-an-esp32-via-node-red-mcu/77171) on Discourse).  Then people can determine quickly what is required to get up and running with it.  And if they want to experiment later on with other sensors, they can search in your nodes overview that redirects them to the page of some node.

Tip: I would remove the *"This effort is intended to evaluate whether it is feasible to support Node-RED on MCUs"* from your readme page.  Let's be honest: this is no evaluation stuff anymore.  Don't be too modest ;-)

**Reply by phoddie** [reply]
> Tip: I would remove the "This effort is intended to evaluate whether it is feasible to support Node-RED on MCUs" from your readme page. Let's be honest: this is no evaluation stuff anymore. Don't be too modest ;-)

Ha! When I originally wrote that, there was no modesty involved. But, indeed the work has evolved and the presentation should reflect that.

> BTW somewhere in the top of your page, I would add a link to a quickstart guide (something similar to our [tutorial](https://discourse.nodered.org/t/tutorial-read-a-sensor-by-running-a-node-red-flow-on-an-esp32-via-node-red-mcu/77171) on Discourse

Agreed.
